hedera-node/src/main/java/com/hedera/services/txns/contract/process/EvmTxProcessor.java

```diff
@@ -114,25 +114,37 @@ public EvmTxProcessor(
 				1);
 	}
 
-	// TODO we can remove the Transaction object
-	protected TransactionProcessingResult execute(Account sender, Optional<Account> receiver, long gasPrice,
-												  long gasLimit, long value, Bytes payload, boolean contractCreation, Instant consensusTime) {
+	protected TransactionProcessingResult execute(Account sender, Address receiver, long gasPrice,
+												  long providedGasLimit, long value, Bytes payload, boolean contractCreation,
+												  Instant consensusTime, boolean isStatic) {
 		try {
+			final long gasLimit = providedGasLimit > dynamicProperties.maxGas() ? dynamicProperties.maxGas() : providedGasLimit;
 			final Wei upfrontCost = Wei.of(Math.addExact(Math.multiplyExact(gasLimit, gasPrice), value));
 			final Gas intrinsicGas =
 					gasCalculator.transactionIntrinsicGasCost(Bytes.EMPTY, contractCreation);
+			final var updater = worldState.updater();
 
-			validateFalse(upfrontCost.compareTo(Wei.of(sender.getBalance())) > 0,
-					ResponseCodeEnum.INSUFFICIENT_ACCOUNT_BALANCE);
-			validateFalse(intrinsicGas.toLong() > gasLimit,
-					ResponseCodeEnum.INSUFFICIENT_GAS);
+			if (!isStatic) {
+				validateFalse(upfrontCost.compareTo(Wei.of(sender.getBalance())) > 0,
+						ResponseCodeEnum.INSUFFICIENT_PAYER_BALANCE);
+				if (intrinsicGas.toLong() > gasLimit) {
+					if (gasLimit < dynamicProperties.maxGas()) {
+						throw new InvalidTransactionException(INSUFFICIENT_GAS);
+					} else {
+						throw new InvalidTransactionException(MAX_GAS_LIMIT_EXCEEDED);
+					}
```

Comment:
Suggesiton -> we could use the following syntax when computing the error message to be thrown:
```Java
throw new InvalidTransactionException(gasLimit < dynamicProperties.maxGas() ? INSUFFICIENT_GAS :
							MAX_GAS_LIMIT_EXCEEDED);
```